<a href="https://colab.research.google.com/github/BenYILDO/breast_cancer_ML/blob/main/MachineLearning.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import seaborn as sns
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import xgboost as xgb
import plotly.express as px

from sklearn.naive_bayes import GaussianNB
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.neighbors import KNeighborsClassifier

from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import (
    f1_score, precision_score, accuracy_score, recall_score,
    balanced_accuracy_score, roc_auc_score, confusion_matrix,
    roc_curve
)



------------



# 1. Veri Setinin Yüklenmesi ve İncelenmesi
## 1.1 Veri Seti Yükleme


Öncelikle yüklenen veri setinin hangi bileşenlerden oluştuğunu görmek için ***keys( )*** fonksiyonu kullanılabilir.




In [ ]:
from sklearn.datasets import load_breast_cancer

data = load_breast_cancer()

data.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])

## 1.2 Veri Çerçevesi Oluşturma

In [ ]:
print("Özellik isimleri:")
print(data.feature_names)

print("\nSınıf isimleri:")
print(data.target_names)

print("\nVeri boyutu:")
print(data.data.shape)

print("\nHedef boyutu:")
print(data.target.shape)

Özellik isimleri:
['mean radius' 'mean texture' 'mean perimeter' 'mean area'
 'mean smoothness' 'mean compactness' 'mean concavity'
 'mean concave points' 'mean symmetry' 'mean fractal dimension'
 'radius error' 'texture error' 'perimeter error' 'area error'
 'smoothness error' 'compactness error' 'concavity error'
 'concave points error' 'symmetry error' 'fractal dimension error'
 'worst radius' 'worst texture' 'worst perimeter' 'worst area'
 'worst smoothness' 'worst compactness' 'worst concavity'
 'worst concave points' 'worst symmetry' 'worst fractal dimension']

Sınıf isimleri:
['malignant' 'benign']

Veri boyutu:
(569, 30)

Hedef boyutu:
(569,)


* data.data: bağımsız değişkenleri içerir
* data.target: hedef sınıfı içerir
* data.feature_names: sütun isimlerini verir
* data.target_names: sınıf isimlerini verir
* data.DESCR: veri setinin açıklamasını içerir



Bu çıktı, veri setinin yalnızca gözlem verilerinden oluşmadığını aynı zamanda hedef değişkeni, özellik isimlerini, sınıf isimlerini ve açıklama metnini de içerdiğini göstermektedir.

In [ ]:
X = pd.DataFrame(data.data, columns=data.feature_names)
y = pd.Series(data.target, name="target")

* Veri seti incelendikten sonra bağımsız değişkenler X,
* hedef değişken ise y olarak ayrılmıştır.
* Özellikler tablo yapısında çalışabilmek için DataFrame, hedef değişken ise tek boyutlu olduğu için Series formatına dönüştürülmüştür.

In [ ]:
print("X boyutu:", X.shape)
print("y boyutu:", y.shape)
print("df boyutu:", df.shape)

X boyutu: (569, 30)
y boyutu: (569,)
df boyutu: (569, 31)


- Veri setinde 569 gözlem ve 30 bağımsız değişken bulunmaktadır.
- Hedef değişken de eklendiğinde toplam sütun sayısı 31 olmaktadır.
- Bu problem, tümörün iyi huylu veya kötü huylu olmasını tahmin etmeye yönelik iki sınıflı bir sınıflandırma problemidir.




